lib/ksuid/activerecord/schema_statements.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module KSUID
+  module SchemaStatements
+    module NativeDatabaseTypes
+      def self.configure
+        env_config = ::ActiveRecord::Base.configurations.configs_for(env_name: Rails.env)
+
+        if env_config.count > 0
+          env_config.each do |c|
+            case c.config["adapter"].to_sym
+            when :postgresql
+              load_postgresql
+            when :sqlite3
+              load_sqlite3
+            when :mysql
+              load_mysql
+            end
+          end
+        else
+          load_postgresql
+          load_sqlite3
+          load_mysql
+        end
+      end
+
+      def self.load_postgresql
+        require "active_record/connection_adapters/postgresql_adapter"
+        ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter::NATIVE_DATABASE_TYPES[:ksuid] = {
```

Comment:
`NATIVE_DATABASE_TYPES` is returned by [native_database_types](https://github.com/rails/rails/blob/v6.0.2.1/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L316-L318) which comes from [abstract adapter](https://github.com/rails/rails/blob/v6.0.2.1/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L13-L18).

`native_database_types` is referenced in the following places:

```
active_record/connection_adapters/abstract_adapter.rb
207:        !native_database_types[type].nil?

active_record/connection_adapters/abstract/schema_statements.rb
16:      def native_database_types
1103:        if native = native_database_types[type]

active_record/connection_adapters/abstract/schema_dumper.rb
62:          limit.inspect if limit && limit != @connection.native_database_types[column.type][:limit]
```

Adding `ksuid` as native type should not have any unwanted side effects.